WordPress/gutenberg
# Block API: Support custom icons for blocks

Closes #1327

This pull request seeks to enhance the block `icon` property to support custom icons, not limiting block implementer to the icons include in the Dashicons set. With these changes, an icon can now be assigned as an element or component.

__Testing instructions:__

Verify tests pass:

```
npm run test-unit blocks/block-icon
```

There is currently no usage of custom icons, but you can apply changes locally and observe effect if you choose. For example:

```diff
diff --git a/blocks/library/image/index.js b/blocks/library/image/index.js
index 739293d9..11a120d0 100644
--- a/blocks/library/image/index.js
+++ b/blocks/library/image/index.js
@@ -23,7 +23,7 @@ const { attr, children } = query;
 registerBlockType( 'core/image', {
        title: __( 'Image' ),
 
-       icon: 'format-image',
+       icon: <Dashicon icon="format-image" />,
 
        category: 'common',
```
